### ISSUE TITLE
temurin@17 and the others are not installing

### ISSUE THREAD

**Issue by adilsoncarvalho** [Homebrew/homebrew-cask#177454]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match / SHA256 mismatch](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error (do not open an issue before trying to open a PR to fix first).

### Description of issue

When trying to install temurin@17 and temurin@21, the installation fails.

I noticed that the packages versions `17.0.11_9` were converted to `17.0.11,9` when generating the path to store the pkg, and it might be failing when creating the directory.

### Command that failed

brew install --verbose --debug --cask temurin@17

### Output of command with `--verbose --debug`

```shell
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading temurin@17
==> Cask::Installer#install
==> Printing caveats
==> Cask::Installer#fetch
==> Downloading https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.11%2B9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.3.6\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 14.5\)\ curl/8.6.0 --header Accept-Language:\ en --retry 3 --fail --location --silent --head https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.11\%2B9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
Already downloaded: /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
==> Checking quarantine support
/usr/bin/env /usr/bin/xattr -h
/usr/bin/env /Library/Developer/CommandLineTools/usr/bin/swift -target arm64-apple-macosx14 /opt/homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Quarantine is available.
==> Verifying Gatekeeper status of /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
==> /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg is quarantined
==> Verifying checksum for 'e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg'
/usr/bin/env hdiutil imageinfo -format /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
==> Installing Cask temurin@17
==> Cask::Installer#stage
==> Extracting primary container
==> Using container class UnpackStrategy::Pkg for /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
/usr/bin/env cp -p -c /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg /opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
==> Verifying Gatekeeper status of /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
==> /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg is quarantined
==> Propagating quarantine from /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg to /opt/homebrew/Caskroom/temurin@17/17.0.11,9
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
/usr/bin/env /usr/bin/xargs -0 -- /bin/chmod -h u+w
/usr/bin/env /usr/bin/xargs -0 -- /usr/bin/xattr -w com.apple.quarantine 0381\;66758538\;\;69FCF695-7CA9-419E-A836-30094AC6F3E3
==> Creating metadata directory: /opt/homebrew/Caskroom/temurin@17/.metadata/17.0.11,9/20240621141123.999
==> Creating metadata subdirectory: /opt/homebrew/Caskroom/temurin@17/.metadata/17.0.11,9/20240621141123.999/Casks
==> Installing artifacts
==> Installing artifact of class Cask::Artifact::Pkg
==> Running installer for temurin@17 with sudo; the password may be necessary.
/usr/bin/sudo -u root -E LOGNAME=acarvalho USER=acarvalho USERNAME=acarvalho -- /usr/sbin/installer -pkg /opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg -target / -verboseR
installer: Error - the package path specified was invalid: '/opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg'.
==> Purging files for version 17.0.11,9 of Cask temurin@17
Error: Failure while executing; `/usr/bin/sudo -u root -E LOGNAME=acarvalho USER=acarvalho USERNAME=acarvalho -- /usr/sbin/installer -pkg /opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg -target / -verboseR` exited with 1. Here's the output:
installer: Error - the package path specified was invalid: '/opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg'.
/opt/homebrew/Library/Homebrew/system_command.rb:352:in `assert_success!'
/opt/homebrew/Library/Homebrew/system_command.rb:75:in `run!'
/opt/homebrew/Library/Homebrew/system_command.rb:40:in `run'
/opt/homebrew/Library/Homebrew/system_command.rb:44:in `run!'
/opt/homebrew/Library/Homebrew/cask/artifact/pkg.rb:63:in `block in run_installer'
/opt/homebrew/Library/Homebrew/cask/artifact/pkg.rb:76:in `with_choices_file'
/opt/homebrew/Library/Homebrew/cask/artifact/pkg.rb:56:in `run_installer'
/opt/homebrew/Library/Homebrew/cask/artifact/pkg.rb:32:in `install_phase'
/opt/homebrew/Library/Homebrew/cask/installer.rb:248:in `block in install_artifacts'
/opt/homebrew/Library/Homebrew/cask/artifact_set.rb:10:in `each'
/opt/homebrew/Library/Homebrew/cask/artifact_set.rb:10:in `each'
/opt/homebrew/Library/Homebrew/cask/installer.rb:241:in `install_artifacts'
/opt/homebrew/Library/Homebrew/cask/installer.rb:113:in `install'
/opt/homebrew/Library/Homebrew/cmd/install.rb:246:in `block in run'
/opt/homebrew/Library/Homebrew/cmd/install.rb:235:in `each'
/opt/homebrew/Library/Homebrew/cmd/install.rb:235:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:92:in `<main>'
```

### Output of `brew doctor` and `brew config`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
  terraform

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
Untap them with `brew untap`.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libcrypto.3.dylib
  /usr/local/lib/libykcs11.2.5.2.dylib
  /usr/local/lib/libykpiv.2.5.2.dylib
  /usr/local/lib/libz.1.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/openssl/aes.h
  /usr/local/include/openssl/asn1.h
  /usr/local/include/openssl/asn1_mac.h
  /usr/local/include/openssl/asn1err.h
  /usr/local/include/openssl/asn1t.h
  /usr/local/include/openssl/async.h
  /usr/local/include/openssl/asyncerr.h
  /usr/local/include/openssl/bio.h
  /usr/local/include/openssl/bioerr.h
  /usr/local/include/openssl/blowfish.h
  /usr/local/include/openssl/bn.h
  /usr/local/include/openssl/bnerr.h
  /usr/local/include/openssl/buffer.h
  /usr/local/include/openssl/buffererr.h
  /usr/local/include/openssl/camellia.h
  /usr/local/include/openssl/cast.h
  /usr/local/include/openssl/cmac.h
  /usr/local/include/openssl/cmp.h
  /usr/local/include/openssl/cmp_util.h
  /usr/local/include/openssl/cmperr.h
  /usr/local/include/openssl/cms.h
  /usr/local/include/openssl/cmserr.h
  /usr/local/include/openssl/comp.h
  /usr/local/include/openssl/comperr.h
  /usr/local/include/openssl/conf.h
  /usr/local/include/openssl/conf_api.h
  /usr/local/include/openssl/conferr.h
  /usr/local/include/openssl/configuration.h
  /usr/local/include/openssl/conftypes.h
  /usr/local/include/openssl/core.h
  /usr/local/include/openssl/core_dispatch.h
  /usr/local/include/openssl/core_names.h
  /usr/local/include/openssl/core_object.h
  /usr/local/include/openssl/crmf.h
  /usr/local/include/openssl/crmferr.h
  /usr/local/include/openssl/crypto.h
  /usr/local/include/openssl/cryptoerr.h
  /usr/local/include/openssl/cryptoerr_legacy.h
  /usr/local/include/openssl/ct.h
  /usr/local/include/openssl/cterr.h
  /usr/local/include/openssl/decoder.h
  /usr/local/include/openssl/decodererr.h
  /usr/local/include/openssl/des.h
  /usr/local/include/openssl/dh.h
  /usr/local/include/openssl/dherr.h
  /usr/local/include/openssl/dsa.h
  /usr/local/include/openssl/dsaerr.h
  /usr/local/include/openssl/dtls1.h
  /usr/local/include/openssl/e_os2.h
  /usr/local/include/openssl/e_ostime.h
  /usr/local/include/openssl/ebcdic.h
  /usr/local/include/openssl/ec.h
  /usr/local/include/openssl/ecdh.h
  /usr/local/include/openssl/ecdsa.h
  /usr/local/include/openssl/ecerr.h
  /usr/local/include/openssl/encoder.h
  /usr/local/include/openssl/encodererr.h
  /usr/local/include/openssl/engine.h
  /usr/local/include/openssl/engineerr.h
  /usr/local/include/openssl/err.h
  /usr/local/include/openssl/ess.h
  /usr/local/include/openssl/esserr.h
  /usr/local/include/openssl/evp.h
  /usr/local/include/openssl/evperr.h
  /usr/local/include/openssl/fips_names.h
  /usr/local/include/openssl/fipskey.h
  /usr/local/include/openssl/hmac.h
  /usr/local/include/openssl/hpke.h
  /usr/local/include/openssl/http.h
  /usr/local/include/openssl/httperr.h
  /usr/local/include/openssl/idea.h
  /usr/local/include/openssl/kdf.h
  /usr/local/include/openssl/kdferr.h
  /usr/local/include/openssl/lhash.h
  /usr/local/include/openssl/macros.h
  /usr/local/include/openssl/md2.h
  /usr/local/include/openssl/md4.h
  /usr/local/include/openssl/md5.h
  /usr/local/include/openssl/mdc2.h
  /usr/local/include/openssl/modes.h
  /usr/local/include/openssl/obj_mac.h
  /usr/local/include/openssl/objects.h
  /usr/local/include/openssl/objectserr.h
  /usr/local/include/openssl/ocsp.h
  /usr/local/include/openssl/ocsperr.h
  /usr/local/include/openssl/opensslconf.h
  /usr/local/include/openssl/opensslv.h
  /usr/local/include/openssl/ossl_typ.h
  /usr/local/include/openssl/param_build.h
  /usr/local/include/openssl/params.h
  /usr/local/include/openssl/pem.h
  /usr/local/include/openssl/pem2.h
  /usr/local/include/openssl/pemerr.h
  /usr/local/include/openssl/pkcs12.h
  /usr/local/include/openssl/pkcs12err.h
  /usr/local/include/openssl/pkcs7.h
  /usr/local/include/openssl/pkcs7err.h
  /usr/local/include/openssl/prov_ssl.h
  /usr/local/include/openssl/proverr.h
  /usr/local/include/openssl/provider.h
  /usr/local/include/openssl/quic.h
  /usr/local/include/openssl/rand.h
  /usr/local/include/openssl/randerr.h
  /usr/local/include/openssl/rc2.h
  /usr/local/include/openssl/rc4.h
  /usr/local/include/openssl/rc5.h
  /usr/local/include/openssl/ripemd.h
  /usr/local/include/openssl/rsa.h
  /usr/local/include/openssl/rsaerr.h
  /usr/local/include/openssl/safestack.h
  /usr/local/include/openssl/seed.h
  /usr/local/include/openssl/self_test.h
  /usr/local/include/openssl/sha.h
  /usr/local/include/openssl/srp.h
  /usr/local/include/openssl/srtp.h
  /usr/local/include/openssl/ssl.h
  /usr/local/include/openssl/ssl2.h
  /usr/local/include/openssl/ssl3.h
  /usr/local/include/openssl/sslerr.h
  /usr/local/include/openssl/sslerr_legacy.h
  /usr/local/include/openssl/stack.h
  /usr/local/include/openssl/store.h
  /usr/local/include/openssl/storeerr.h
  /usr/local/include/openssl/symhacks.h
  /usr/local/include/openssl/thread.h
  /usr/local/include/openssl/tls1.h
  /usr/local/include/openssl/trace.h
  /usr/local/include/openssl/ts.h
  /usr/local/include/openssl/tserr.h
  /usr/local/include/openssl/txt_db.h
  /usr/local/include/openssl/types.h
  /usr/local/include/openssl/ui.h
  /usr/local/include/openssl/uierr.h
  /usr/local/include/openssl/whrlpool.h
  /usr/local/include/openssl/x509.h
  /usr/local/include/openssl/x509_vfy.h
  /usr/local/include/openssl/x509err.h
  /usr/local/include/openssl/x509v3.h
  /usr/local/include/openssl/x509v3err.h
  /usr/local/include/ykpiv/ykpiv-config.h
  /usr/local/include/ykpiv/ykpiv.h
  /usr/local/include/zlib.h

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/ykcs11.pc
  /usr/local/lib/pkgconfig/ykpiv.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libykcs11.a
  /usr/local/lib/libykpiv.a
```

### Output of `brew tap`

```shell
homebrew/cask-versions
```

**Comment by razvanazamfirei** [comment]
What is the output of `brew info temurin@17`? Can you run `brew untap homebrew/cask-versions` and try again?

**Comment by adilsoncarvalho** [comment]
Sure thing. Here they are:

```
> brew untap homebrew/cask-versions
Untapping homebrew/cask-versions...
Untapped (38 files, 79.4MB).
```

```
> brew info temurin@17
==> temurin@17: 17.0.11,9
https://adoptium.net/
Not installed
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/t/temurin@17.rb
==> Name
Eclipse Temurin Java Development Kit
==> Description
JDK from the Eclipse Foundation (Adoptium)
==> Artifacts
OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg (Pkg)
==> Analytics
install: 7,072 (30 days), 13,668 (90 days), 13,669 (365 days)
```

And I tried again to install. Same result.

```
> brew install temurin@17
==> Downloading https://formulae.brew.sh/api/formula.jws.json
############################################################################################################## 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
############################################################################################################## 100.0%
==> Downloading https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.11%2B9/OpenJDK17U-jdk_aarch
Already downloaded: /Users/acarvalho/Library/Caches/Homebrew/downloads/e2ce3344fda7c421751d0e95669ec661b29c3edb46c29aa3edac482f55164a41--OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg
==> Installing Cask temurin@17
==> Running installer for temurin@17 with sudo; the password may be necessary.


Launch with elevated privileges

Password: 
installer: Error - the package path specified was invalid: '/opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg'.
==> Purging files for version 17.0.11,9 of Cask temurin@17
Error: Failure while executing; `/usr/bin/sudo -u root -E LOGNAME=acarvalho USER=acarvalho USERNAME=acarvalho -- /usr/sbin/installer -pkg /opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg -target /` exited with 1. Here's the output:
installer: Error - the package path specified was invalid: '/opt/homebrew/Caskroom/temurin@17/17.0.11,9/OpenJDK17U-jdk_aarch64_mac_hotspot_17.0.11_9.pkg'.
```

**Comment by khipp** [comment]
You are using Homebrew version 4.3.6. A recent change in file copying (https://github.com/Homebrew/brew/pull/17373) was included in this release, which caused installation issues for some users. This change has since been reverted.

The fix has not yet been officially released, but you can resolve the issue by running `brew update-reset` before trying again. This command will reset your Homebrew installation to track the latest changes.

**Comment by MikeMcQuaid** [comment]
This is released/fixed in Homebrew 4.3.7 now: https://github.com/Homebrew/brew/releases/tag/4.3.7
